lib/galaxy/tools/parameters/basic.py

```diff
@@ -1319,24 +1327,22 @@ def recurse_options(legal_values, options):
 
     def from_json(self, value, trans, other_values={}):
         legal_values = self.get_legal_values(trans, other_values)
-        if len(list(legal_values)) == 0 and trans.workflow_building_mode:
+        if not legal_values:
```

Comment:
Should't we handle `workflow_building_mode` as at lines 862-864?

Reply:
Yes, we should. I wanted to do that in a follow-up applying the same rules established here for select fields to dropdowns.